File: src/main/java/renankummer/udemy/springwebapp/controller/AuthorController.java
package renankummer.udemy.springwebapp.controller;

import lombok.AllArgsConstructor;
import lombok.NonNull;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import renankummer.udemy.springwebapp.repository.AuthorRepository;

@Controller
@AllArgsConstructor
public class AuthorController {

    private final AuthorRepository repository;

    @GetMapping("/authors")
    public String getAuthors(@NonNull Model model) {
        model.addAttribute("authors", repository.findAll());
        return "authors/list";
    }
}

File: src/main/kotlin/renankummer/udemy/springwebapp/controller/AuthorController.kt
package renankummer.udemy.springwebapp.controller

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.ui.set
import org.springframework.web.bind.annotation.GetMapping
import renankummer.udemy.springwebapp.repository.AuthorRepository

@Controller
class AuthorController(val authorRepository: AuthorRepository) {

    @GetMapping("/authors")
    fun getAuthors(model: Model): String {
        model["authors"] = authorRepository.findAll()
        return "authors/list"
    }
}
